src/sentry/event_manager.py

```diff
@@ -897,36 +893,36 @@ def _get_or_create_release_many(jobs: Sequence[Job], projects: ProjectsMapping)
                         "event_manager.dynamic_sampling_observe_latest_release"
                     ) as metrics_tags:
                         try:
-                            environment = _extract_latest_release_data(data)
-                            release_observed_in_last_24h = observe_release(
-                                project_id, release.id, environment
+                            latest_release_params = LatestReleaseParams(
+                                release=release,
+                                project=projects[project_id],
+                                environment=_get_environment_from_transaction(data),
                             )
-                            if not release_observed_in_last_24h:
-                                span.set_tag(
-                                    "dynamic_sampling.observe_release_status",
-                                    f"New release observed {release.id}",
+
+                            def on_release_boosted() -> None:
+                                message = (
+                                    f"New (release, environment) pair ({latest_release_params.release.id}, "
+                                    f"{latest_release_params.environment}) observed and boosted"
```

Comment:
We should not set high-cardinality tags like this in either sentry or statsd. I would remove it from the `metrics.timer` and use `set_data` on the span instead.

Reply:
Will do! Thanks for the suggestion